src/coreclr/jit/lower.cpp

```diff
@@ -5133,41 +5133,9 @@ GenTree* Lowering::LowerAdd(GenTreeOp* node)
 // Returns:
 //    The next node to lower.
 //
-GenTree* Lowering::LowerBinaryArithmetic(GenTreeOp* node)
+GenTree* Lowering::LowerBinaryArithmeticCommon(GenTreeOp* node)
 {
-    // TODO-CQ-XArch: support BMI2 "andn" in codegen and condition
-    // this logic on the support for the instruction set on XArch.
-    CLANG_FORMAT_COMMENT_ANCHOR;
```

Comment:
The current implementation was intended to be arm specific so I moved it there. If someone implement xarch andn and finds the code is identical it can be moved up to the common location but for now it feels cleaner to have something that's arm specific in the arm specific file than it does to have if ifdefed for arm in the common file.

Reply:
> The current implementation was intended to be arm specific so I moved it there

I wrote that code so can say with confidence it was very intentional that it lives in the common function :).

If we are to separate them however, as it is true this current implementation is ARMARCH-only, `LowerBinaryArithmeticCommon` needs to be renamed back to `LowerBinaryArithmetic` and deleted (as it now does not do anything but call the arch-specific function). Also the `TODO-CQ-XArch: support BMI2 "andn"..."` needs to mention this ARMARCH logic.

Reply:
If I move it back up to common I have the problem that I need to know if the node has been replaced and containment checked by the hwintrinsic function for that purpose or I need have arm one path on arm and one path on xarch which doesn't make any sense to do in the common file when there's a file for each one of those things. I suppose this is why people seem to change nodes not replace them.

Reply:
Dealt with.